internal/clients/team1/iifo.go

```diff
@@ -56,3 +71,138 @@ func (c *client) ReceiveForageInfo(forageInfos []shared.ForageShareInfo) {
 			)
 	}
 }
+
+/******************************/
+/*** 		 Disasters 		  */
+/******************************/
+
+// findConfidence is only called when a disaster has happened. Therefore len(disasterHistory) > 0
+func (c client) findConfidence() float64 {
+	disasterHistory := c.disasterInfo.allDisaster
+	meanDisaster := c.disasterInfo.meanDisaster
+	totalDisaster := disasters.DisasterReport{}
+	for _, disaster := range disasterHistory {
+		totalDisaster.X += math.Pow(disaster.X-meanDisaster.X, 2)
+		totalDisaster.Y += math.Pow(disaster.Y-meanDisaster.Y, 2)
+		totalDisaster.Magnitude += math.Pow(disaster.Magnitude-meanDisaster.Magnitude, 2)
+	}
+
+	disasterHistorySize := float64(len(disasterHistory))
+	sqrtDisasterHistory := math.Sqrt(disasterHistorySize)
+	xSD := math.Sqrt(totalDisaster.X / disasterHistorySize)
+	ySD := math.Sqrt(totalDisaster.Y / disasterHistorySize)
```

Comment:
div0

Reply:
below too

Reply:
Never gonna be a disasterHistorySize == 0 but i'll add an if statement.